openmc/data/njoy.py

```diff
@@ -0,0 +1,593 @@
+{mat} {temperature}
+1 1/
+/
+"""
+
+_THERMAL_TEMPLATE_THERMR = """
+thermr / %%%%%%%%%%%%%%%% Add thermal scattering data (free gas) %%%%%%%%%%%%%%%
+0 {nthermr1_in} {nthermr1}
+0 {mat} 12 {num_temp} 1 0 {iform} 1 221 1/
+{temps}
+{error} {energy_max}
+thermr / %%%%%%%%%%%%%%%% Add thermal scattering data (bound) %%%%%%%%%%%%%%%%%%
+{nthermal_endf} {nthermr2_in} {nthermr2}
+{mat_thermal} {mat} 16 {num_temp} {inelastic} {elastic} {iform} {natom} 222 1/
+{temps}
+{error} {energy_max}
+"""
+
+_THERMAL_TEMPLATE_ACER = """
+acer / %%%%%%%%%%%%%%%%%%%%%%%% Write out in ACE format %%%%%%%%%%%%%%%%%%%%%%%%
+{nendf} {nthermal_acer_in} 0 {nace} {ndir}
+2 0 1 .{ext}/
+'{library}: {zsymam_thermal} processed by NJOY'/
+{mat} {temperature} '{data.name}' /
+{zaids} /
+222 64 {mt_elastic} {elastic_type} {data.nmix} {energy_max} {iwt}/
+"""
+
+
+def run(commands, tapein, tapeout, input_filename=None, stdout=False,
+        njoy_exec='njoy'):
+    """Run NJOY with given commands
+
+    Parameters
+    ----------
+    commands : str
+        Input commands for NJOY
+    tapein : dict
+        Dictionary mapping tape numbers to paths for any input files
+    tapeout : dict
+        Dictionary mapping tape numbers to paths for any output files
+    input_filename : str, optional
+        File name to write out NJOY input commands
+    stdout : bool, optional
+        Whether to display output when running NJOY
+    njoy_exec : str, optional
+        Path to NJOY executable
+
+    Raises
+    ------
+    subprocess.CalledProcessError
+        If the NJOY process returns with a non-zero status
+
+    """
+
+    if input_filename is not None:
+        with open(str(input_filename), 'w') as f:
+            f.write(commands)
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        # Copy evaluations to appropriates 'tapes'
+        for tape_num, filename in tapein.items():
+            tmpfilename = os.path.join(tmpdir, 'tape{}'.format(tape_num))
+            shutil.copy(str(filename), tmpfilename)
+
+        # Start up NJOY process
+        njoy = Popen([njoy_exec], cwd=tmpdir, stdin=PIPE, stdout=PIPE,
+                     stderr=STDOUT, universal_newlines=True)
+
+        njoy.stdin.write(commands)
+        njoy.stdin.flush()
+        lines = []
+        while True:
+            # If process is finished, break loop
+            line = njoy.stdout.readline()
+            if not line and njoy.poll() is not None:
+                break
+
+            lines.append(line)
+            if stdout:
+                # If user requested output, print to screen
+                print(line, end='')
+
+        # Check for error
+        if njoy.returncode != 0:
+            raise CalledProcessError(njoy.returncode, njoy_exec,
+                                     ''.join(lines))
+
+        # Copy output files back to original directory
+        for tape_num, filename in tapeout.items():
+            tmpfilename = os.path.join(tmpdir, 'tape{}'.format(tape_num))
```

Comment:
Python has a builtin library for tempfiles like this if you want to simplify things: https://docs.python.org/3/library/tempfile.html